jax/_src/numpy/fft.py

```diff
@@ -400,15 +400,133 @@ def ifft(a: ArrayLike, n: int | None = None,
+      of the array along ``axis`` is ``(n/2)+1``, if ``n`` is even and ``(n+1)/2``,
+      if ``n`` is odd. If not specified, ``n`` defaults to the dimension of ``a``
+      along ``axis``.
+    axis: int, default=-1. Specifies the axis along which the transform is computed.
+      If not specified, the transform is computed along axis -1.
+    norm: string. The normalization mode. "backward", "ortho" and "forward" are
+      supported.
+
+  Returns:
+    An array containing the one-dimensional discrete Fourier transform of ``a``.
+    The dimension of the array along ``axis`` is ``(n/2)+1``, if ``n`` is even and
+    ``(n+1)/2``, if ``n`` is odd.
+
+  See also:
+    - :func:`jax.numpy.fft.fft`: Computes a one-dimensional discrete Fourier
+      transform.
+    - :func:`jax.numpy.fft.irfft`: Computes a one-dimensional inverse discrete
+      Fourier transform for real input.
+    - :func:`jax.numpy.fft.rfftn`: Computes a multidimensional discrete Fourier
+      transform for real input.
+    - :func:`jax.numpy.fft.irfftn`: Computes a multidimensional inverse discrete
+      Fourier transform for real input.
+
+  Examples:
+    ``jnp.fft.rfft`` computes the transform along ``axis -1`` by default.
+
+    >>> x = jnp.array([[1, 3, 5],
+    ...                [2, 4, 6]])
+    >>> jnp.fft.irfft(x)
+    Array([[ 3., -1.,  0., -1.],
+           [ 4., -1.,  0., -1.]], dtype=float32)
+
+    When ``n=3``, dimension of the transform along axis -1 will be ``(3+1)/2 =2``
+    and dimension along other axes will be the same as that of input.
+
+    >>> with jnp.printoptions(precision=2, suppress=True):
+    ...   print(jnp.fft.rfft(x, n=3))
+    [[ 9.+0.j   -3.+1.73j]
+     [12.+0.j   -3.+1.73j]]
+
+    When ``n=4`` and ``axis=0``, dimension of the transform along ``axis 0`` will
+    be ``(4/2)+1 =3`` and dimension along other axes will be same as that of input.
+
+    >>> with jnp.printoptions(precision=2, suppress=True):
+    ...   print(jnp.fft.rfft(x, n=4, axis=0))
+    [[ 3.+0.j  7.+0.j 11.+0.j 15.+0.j]
+     [ 1.-2.j  3.-4.j  5.-6.j  7.-8.j]
+     [-1.+0.j -1.+0.j -1.+0.j -1.+0.j]]
+
+    ``jnp.fft.ifft`` can be used to reconstruct ``x`` from the result of
+    ``jnp.fft.fft``.
+
+    >>> x_rfft = jnp.fft.rfft(x)
+    >>> jnp.allclose(x, jnp.fft.irfft(x_rfft))
+    Array(True, dtype=bool)
+  """
   return _fft_core_1d('rfft', xla_client.FftType.RFFT, a, n=n, axis=axis,
                       norm=norm)
 
-@implements(np.fft.irfft)
+
 def irfft(a: ArrayLike, n: int | None = None,
           axis: int = -1, norm: str | None = None) -> Array:
+  r"""Compute a one-dimensional inverse discrete Fourier transform for real input.
+
+  JAX implementation of :func:`numpy.fft.irfft`.
+
+  Args:
+    a: real-valued input array.
+    n: int. Specifies the dimension of the result along ``axis``. If not specified,
+      the dimension of the result along ``axis`` is ``2*(m-1)``, where ``m`` is the
```

Comment:
```suggestion
      the dimension of the result along ``axis`` is ``n = 2*(m-1)``, where ``m`` is the
```

It might be worth adding this ⬆️ or something similar because I got confused about the relationship between `n` and `m`.

Reply:
Changed